src/resetUsedTokens.test.ts

```diff
@@ -0,0 +1,89 @@
+import { describe, expect, test } from "bun:test";
+import { Redis } from "@upstash/redis";
+import { MultiRegionRatelimit } from "./multi";
+import type { Ratelimit } from "./ratelimit";
+import { RegionRatelimit } from "./single";
+import { Algorithm, Context, MultiRegionContext, RegionContext } from "./types";
+
+const limit = 10;
+const refillRate = 10;
+const windowString = "30s";
+
+function run<TContext extends Context>(builder: Ratelimit<TContext>) {
+  const id = crypto.randomUUID();
+
+  describe("resetUsedTokens", () => {
+    test("reset the tokens", async () => {
+      // Consume tokens until the remaining tokens are either equal to 2 or lesser than that
+
+      for (let i = 0; i < 15; i++) {
+        await builder.limit(id);
+      }
+
+      // reset tokens
+      await builder.resetUsedTokens(id);
+      setTimeout(async () => {
+        const { remaining } = await builder.limit(id);
+        expect(remaining).toBe(limit - 1);
+      }, 20000);
```

Comment:
this test doesn't work. `expect(remaining).toBe(limit - 1)` is not checked properly. When I try `expect(remaining).toBe(123123)`, it still passes. Is there a reason why we used `setTimeout`?

Reply:
Yes I have add few changes here.